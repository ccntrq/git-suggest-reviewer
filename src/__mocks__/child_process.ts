/* eslint-disable @typescript-eslint/no-explicit-any */
import {jest} from '@jest/globals';

const child_process = jest.createMockFromModule('child_process') as any;

let mockThrow = false;
function __setMockThrow(value: boolean): void {
  mockThrow = value;
}

function execSync(...args: any[]): any[] {
  if (mockThrow) {
    throw new Error('MockError');
  }
  return args;
}

child_process.__setMockThrow = __setMockThrow;
child_process.execSync = execSync;

module.exports = child_process;
