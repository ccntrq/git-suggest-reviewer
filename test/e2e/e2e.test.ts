import {describe, expect, test} from '@jest/globals';
import {spawnSync} from 'child_process';

describe('e2e test git-suggest-reviewer', () => {
  test('prints usage with --help', () => {
    const result = spawnSync('node', ['./dist/cli.js', '--help'], {
      encoding: 'utf-8',
    });

    expect(result.status).toBe(0);
    expect(result.stdout).toMatchSnapshot();
    expect(result.stderr).toEqual('');
  });
  test('prints usage to stderr and sets exitcode to 1 with no arguments', () => {
    const result = spawnSync('node', ['./dist/cli.js'], {
      encoding: 'utf-8',
    });

    expect(result.status).toBe(1);
    expect(result.stdout).toEqual('');
    expect(result.stderr).toMatchSnapshot();
  });
});
