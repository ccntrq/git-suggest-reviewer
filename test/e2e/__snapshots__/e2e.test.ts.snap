// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e test git-suggest-reviewer prints usage to stderr and sets exitcode to 1 with no arguments 1`] = `
"git-suggest-reviewers

Suggest candidates for a code review based on git history.

USAGE:
  git-suggest-reviewers [OPTIONS]
  git-suggest-reviewers [OPTIONS] base_revision

Options:
    -h --help       Show this help
    -v --version    Print program version
    --verbose       Be more verbose

"
`;

exports[`e2e test git-suggest-reviewer prints usage with --help 1`] = `
"git-suggest-reviewers

Suggest candidates for a code review based on git history.

USAGE:
  git-suggest-reviewers [OPTIONS]
  git-suggest-reviewers [OPTIONS] base_revision

Options:
    -h --help       Show this help
    -v --version    Print program version
    --verbose       Be more verbose

"
`;
